### PR TITLE
Update index.md

### PR DESCRIPTION
**Pull request by mikesmeltzer**
Add breakdown of the event system with code examples for some of the categories.

### CODE REVIEW

File: content/features/extensibility/event-system/index.md [hunk 1]
```diff
@@ -34,7 +113,230 @@ You can attach code to the following Container events:
...
+    
+            }
+    
+        }
+    }
+
+
+In the above example, we are registering a Skin Event Listener called **SampleContainerEventListener.OnContainerLoad**. This is a static class and method that defines what code will run when the Container OnLoad event triggers. 
+
+Since all modules on a page in a DNN website are wrapped in a container, we can assume that this code will run when every container loads on every page within a DNN website.
```
Comment (bdukes) [inline content/features/extensibility/event-system/index.md:155]: Does this run if the _Display Container_ tab-module setting isn't checked?

File: content/features/extensibility/event-system/index.md [hunk 2]
```diff
@@ -9,23 +9,102 @@ dnnversion:
...
+
+    using System;
+    using DotNetNuke.Services.Exceptions;
+    using DotNetNuke.UI.Skins.EventListeners;
+    
+    namespace SampleProject.DnnEvents.SkinEvents
+    {
+        public static class SampleSkinEventListener
+        {
+            public static void OnSkinLoad(object sender, SkinEventArgs e)
```
Comment (bdukes) [inline content/features/extensibility/event-system/index.md:89]: Are there any interesting properties of `SkinEventArgs` that it's worth calling out?

File: content/features/extensibility/event-system/index.md [hunk 3]
```diff
@@ -34,7 +113,230 @@ You can attach code to the following Container events:
...
+You can attach code to the following User events: 
+
+ - UserAuthenticated
+ - UserCreated 
+ - UsserRemoved (deleted from user manager) 
+ - UserDeleted (deleted from recycle bin) 
+ - UserApproved (restored from the page recycle bin)
+ - UserUpdated
+
+User Events are registered by decorating a class with [Export(typeof(IUserEventHandlers))] and implementing the IUserEventHandlers interface.
```
Comment (bdukes) [inline content/features/extensibility/event-system/index.md:273]:
```suggestion
User Events are registered by decorating a class with `[Export(typeof(IUserEventHandlers))]` and implementing the [`IUserEventHandlers`](xref:DotNetNuke.Entities.Users.IUserEventHandlers) interface.
```

File: content/features/extensibility/event-system/index.md [hunk 4]
```diff
@@ -34,7 +113,230 @@ You can attach code to the following Container events:
...
+You can attach code to the following Tab events: 
+
+ - TabCreated
+ - TabUpdated 
+ - TabRemoved (deleted from page manager) 
+ - TabDeleted (deleted from page recycle bin) 
+ - TabRestored (restored from the page recycle bin)
+ - TabMarkedAsPublished
+
+Tab Events are registered by decorating a class with [Export(typeof(ITabEventHandler))] and implementing the ITabEventHandler interface.
```
Comment (bdukes) [inline content/features/extensibility/event-system/index.md:210]:
```suggestion
Tab Events are registered by decorating a class with `[Export(typeof(ITabEventHandler))]` and implementing the [`ITabEventHandler`](xref:DotNetNuke.Entities.Tabs.Actions.ITabEventHandler) interface.
```

**Review by bdukes — COMMENTED** [review]
Thanks!

File: content/features/extensibility/event-system/index.md (hunk 1 above)
Comment (mikesmeltzer) [reply content/features/extensibility/event-system/index.md:155]: I haven't tested but no I don't believe it does. These are triggered from the container base class, if the container isn't visible on the page then the container code should not trigger for that specific module/container combo.

File: content/features/extensibility/event-system/index.md (hunk 2 above)
Comment (mikesmeltzer) [reply content/features/extensibility/event-system/index.md:89]: If it's Ok with you, I'll make a note on my end to look to see if I can provide some additional detail on my second pass through the page as I build it out a bit more. For now, I think they are self-explanatory for the audience that will be using them at this point.  As well, I'll try to come up with a supporting blog post about how/why a dev might use these? I can pull together my info from Southern Fried that I did the other week on this topic which had real world examples.

**Comment by mikesmeltzer** [comment]
Sorry @bdukes I didn't get email alerts for your comments for some reason so just seeing them now.

**Review by david-poindexter — APPROVED** [review]
Awesome - thanks @mikesmeltzer and @bdukes 🎉

**Pull request merged**